code/common/finspace.q

```diff
@@ -41,3 +41,10 @@ notifyhdb:{[cluster;changeset]
       .aws.update_kx_cluster_databases[string[cluster];.aws.sdbs[.aws.db[.finspace.database;changeset[`id];.aws.cache["CACHE_1000";"/"]]];.aws.sdep["NO_RESTART"]]
       // TODO - Also need to figure out the ideal logic if a changeset fails to create. Possibly recreate and re-run notifyhd
    }
+
+// function to close connection to TP and remove unwanted data in WDB and RDB's 
+eopdatacleanup:{[dict]
+    hclose each exec w from .servers.SERVERS where proctype = `segmentedtickerplant;
+    // function to parse icounts dict and remove all data after a given index for RDB and WDB's 
+    {[sym;ind]delete from sym where i >= ind}'[key dict;first each value dict]
```

Comment:
why is the `first each` needed? is it not a single value for each table?

I also think it's confusing to use `sym` as the argument in this lambda, that's the name of a table, right? so something like `t` or `table` would probably be more clear?

Reply:
From my testing i saw that the `value dictionaryname` enlists each keyed entry, hence the `first each` as we are itterating through the list of lists 

will update the `sym` var to `t` as per your suggestion